hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java

```diff
@@ -369,7 +369,8 @@ private boolean skipForMerge(final RegionStates regionStates, final RegionInfo r
       }
       final long currentSizeMb = getRegionSizeMB(current);
       final long nextSizeMb = getRegionSizeMB(next);
-      if (currentSizeMb + nextSizeMb < avgRegionSizeMb) {
+      // always merge away empty regions when they present themselves.
+      if (currentSizeMb == 0 || nextSizeMb == 0 || currentSizeMb + nextSizeMb < avgRegionSizeMb) {
```

Comment:
There is a special case for size to be 0, wondering if the region size is very small but nonzero, and there is a large size neighbor region, can we do something about it? 
100 1 100  (avg size is 20), in this case, 1 wont be merged as well.  What if some logic like

currentSizeMb < 0.1 * avgRegionSizeMb and currentSizeMb/nextSizeMb < 0.1 (switch currentSizeMb and nextSizeMb as well), then merge

Reply:
I like this idea of a fuzzy threshold idea. What if we merge a little more aggressively, expressed relative to `avgRegionSizeMb`? Something like

```
if (currentSizeMb + nextSizeMb < avgRegionSizeMb * 0.4) {...}
```

This gives us a strong preference toward larger regions, with a threshold based on the average size.

I guess next you'll same "make it configurable" :)

Reply:
> ```
> if (currentSizeMb + nextSizeMb < avgRegionSizeMb * 0.4) {...}
> ```

The above statement does not address the case that 100 1 80 (average 60), in this case 100 is not qualified for split, and none is qualified for merge as well.

Do not mean to block your commit, and for cases like above, need to spend more time to find a right solution, we can create an improvement jira to address it.

Reply:
@huaxiangsun I've been thinking about this and haven't come to a formula that I like. So let me propose this: rather than basing this decision on a scaling factor, how about we decide on some arbitrary value to use as meaning "effectively 0" in size. Say, any region <= 10mb will be considered as "effectively 0-sized" and we'll merge it into it's neighbor when possible. I supposed we can make this configurable as well...

What do you think?

Reply:
Yeah, this is a good idea!

Reply:
Argh. This idea is in direct contradiction to the feature implemented by "The minimum size for a region to be considered for a merge, in whole MBs."

```
  static final String MERGE_MIN_REGION_SIZE_MB_KEY = "hbase.normalizer.merge.min_region_size.mb";
  static final int DEFAULT_MERGE_MIN_REGION_SIZE_MB = 1;
```

Reply:
I think what we need is to unset the default values of `hbase.normalizer.merge.min_region_age.days` and `hbase.normalizer.merge.min_region_size.mb`. If someone wants the features, make them opt-in.

Reply:
Oh, and `hbase.normalizer.min.region.count` as well.

Reply:
There hasn't been much feedback on the DISCUSS thread.

I'm of the mind that we can resolve this specific bug using the special-case (`if == 0`) code I have in this patch. I think we should return with another change that allows the `min_*` values to be unset, and probably remove the defaults for everything except `min_region_age.days`.

Reply:
I think this is good enough considering if region is empty it is in KB's as per my experience(i can be wrong here). One issue is if there are few rows which are yet to expire then the probability of next region to be more than avgRegionSize is also low but ofcourse possible. Silver lline is eventually we will merge it maybe in 2-3 weeks.

Reply:
Replying to my earlier comment,

> So let me propose this: rather than basing this decision on a scaling factor, how about we decide on some arbitrary value to use as meaning "effectively 0" in size.

Per my above, and @mnpoonia 's, we've already decided that "effectively 0" means "less than 1mb according to reported metrics." Thus I think we should keep this "special logic".